From c08b586323ef3e2d305ea3cca5021e148b1945a8 Mon Sep 17 00:00:00 2001
Message-Id: <c08b586323ef3e2d305ea3cca5021e148b1945a8.1586875951.git.mjg@fedoraproject.org>
From: Michael J Gruber <mjg@fedoraproject.org>
Date: Tue, 14 Apr 2020 16:50:37 +0200
Subject: [PATCH] fix duplicate case fix

41cda39 ("silencing compiler warning", 2019-01-04) tried to fix the
duplicate case warning, but this fails if the expression cannot be
evaluated by the preprocessor.

Fedora regex has POSIX egrep, so avoid the if.

Signed-off-by: Michael J Gruber <mjg@fedoraproject.org>
---
 rewrite.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/rewrite.c b/rewrite.c
index d34ccd1..c2a6342 100644
--- a/rewrite.c
+++ b/rewrite.c
@@ -1360,9 +1360,6 @@ char* get_regex_syntax()			   /*                        */
   { case RE_SYNTAX_EMACS:       return "emacs";	   /*                        */
     case RE_SYNTAX_AWK:         return "awk";	   /*                        */
     case RE_SYNTAX_GREP:        return "grep";	   /*                        */
-#if (RE_SYNTAX_EGREP != RE_SYNTAX_POSIX_EGREP)
-    case RE_SYNTAX_EGREP:       return "egrep";	   /*                        */
-#endif
     case RE_SYNTAX_POSIX_AWK:   return "posix_awk";/*                        */
     case RE_SYNTAX_POSIX_EGREP: return "posix_egrep";/*                      */
     case RE_SYNTAX_SED:         return "sed";	   /*                        */
-- 
2.26.0.280.gaba8734f63

